pkg/controllers/provisioning/v1alpha1/reallocator/filter.go

```diff
@@ -25,40 +28,75 @@ type Filter struct {
 	kubeClient client.Client
 }
 
-// Gets Nodes fitting some underutilization
-// TODO: add predicate
-func (f *Filter) GetUnderutilizedNodes(ctx context.Context) ([]*v1.Node, error) {
-	nodeList := &v1.NodeList{}
+// Gets Nodes fitting some underutilization predicate
+func (f *Filter) GetUnderutilizedNodes(ctx context.Context, provisioner *v1alpha1.Provisioner) ([]*v1.Node, error) {
 	underutilized := []*v1.Node{}
 
-	// 1. Get all nodes
-	if err := f.kubeClient.List(ctx, nodeList); err != nil {
-		return nil, fmt.Errorf("listing nodes, %w", err)
+	// 1. Get all provisioner labeled nodes
+	nodeList, err := f.getProvisionerNodes(ctx, provisioner)
+	if err != nil {
+		return nil, fmt.Errorf("listing underutilized nodes, %w", err)
 	}
 
 	// 2. Get nodes and the pods on each node
 	for _, node := range nodeList.Items {
 		pods, err := f.getPodsOnNode(ctx, node.Name)
 		if err != nil {
-			return []*v1.Node{}, fmt.Errorf("filtering pods on nodes, %w", err)
+			return []*v1.Node{}, fmt.Errorf("listing pods on underutlized provisioner nodes, %w", err)
 		}
-		if f.isUnderutilized(&node, pods) {
+		// Only checks if it has 0 non-daemon pods right now
+		if f.isUnderutilized(pods) {
 			underutilized = append(underutilized, &node)
 		}
 	}
 	return underutilized, nil
 }
 
+func (f *Filter) GetExpiredNodes(ctx context.Context, provisioner *v1alpha1.Provisioner) ([]*v1.Node, error) {
+	nodeList, err := f.getProvisionerNodes(ctx, provisioner)
+	if err != nil {
+		return nil, fmt.Errorf("listing expired TTLed nodes, %w", err)
```

Comment:
Got it. I wanted to use the error annotation here to describe that it's listing nodes, but in the `GetExpiredNodes`. This might not be the place to do this. Thanks.